subsys/math/CMakeLists.txt

```diff
@@ -0,0 +1,8 @@
+# Copyright (c) 2022 Google LLC
+# SPDX-License-Identifier: Apache-2.0
+
+if(CONFIG_CMSIS_DSP)
+  if(NOT CONFIG_HAS_CMSIS_CORE)
+    zephyr_include_directories(public)
+  endif()
+endif()
```

Comment:
What about dsp.h (with the zdsp_ function definitions identical to CMSIS-DSP)  multiple backends support?

Reply:
I think I might be confused. Your comment at https://github.com/zephyrproject-rtos/zephyr/pull/51059#pullrequestreview-1133991645 made it seem like you didn't want to have our own custom wrappers.

> So, unless you plan to document all these functions and add full comprehensive test suites for all of them, please keep the API identical to that of the CMSIS-DSP.
>
> Also, remember that the functions you have implemented here is just a subset of the "Basic Math" functions -- there are hundreds more in the CMSIS-DSP and deviating from the CMSIS-DSP is not going to be a sustainable approach unless we have someone working on this full time.

Does this mean that you would like to have our own functions with CMSIS-DSP as a backend?

Reply:
> I think I might be confused. Your comment at [#51059 (review)](https://github.com/zephyrproject-rtos/zephyr/pull/51059#pullrequestreview-1133991645) made it seem like you didn't want to have our own custom wrappers.

All I said was to "keep the API identical to that of the CMSIS-DSP", not to get rid of the Zephyr DSP API layer altogether.

> Does this mean that you would like to have our own functions with CMSIS-DSP as a backend?

Yes, at least, that was the compromise and what we settled on in the chat:

> the idea (discussed above) is to define Zephyr DSP API that is a direct 1-to-1 mapping to CMSIS-DSP
> so it will be basically CMSIS-DSP API, but with zdsp_ prefix.